ui/app/templates/jobs/run.hbs

```diff
@@ -0,0 +1,82 @@
+    </div>
+  {{/if}}
+  {{#if planError}}
+    <div data-test-plan-error class="notification is-danger">
+      <h3 class="title is-4" data-test-plan-error-title>Plan Error</h3>
+      <p data-test-plan-error-message>{{planError}}</p>
+    </div>
+  {{/if}}
+  {{#if runError}}
+    <div data-test-run-error class="notification is-danger">
+      <h3 class="title is-4" data-test-run-error-title>Run Error</h3>
+      <p data-test-run-error-message>{{runError}}</p>
+    </div>
+  {{/if}}
+
+  {{#if (eq stage "editor")}}
+    {{#if showEditorMessage}}
+      <div class="notification is-info">
+        <div class="columns">
+          <div class="column">
+            <h3 class="title is-4" data-test-editor-help-title>Run a Job</h3>
+            <p data-test-editor-help-message>Paste or author HCL or JSON to submit to your cluster. A plan will be requested before the job is submitted.</p>
+          </div>
+          <div class="column is-centered is-minimum">
+            <button class="button is-info" onclick={{toggle-action "showEditorMessage" this}} data-test-editor-help-dismiss>Okay</button>
+          </div>
+        </div>
+      </div>
+    {{/if}}
+    <div class="boxed-section">
+      <div class="boxed-section-head">
+        Job Definition
+      </div>
+      <div class="boxed-section-body is-full-bleed">
+        {{ivy-codemirror
+          data-test-editor
+          value=(or model._newDefinition jobSpec)
+          valueUpdated=(action (mut model._newDefinition))
+          options=(hash
+            mode="javascript"
+            theme="hashi"
+            tabSize=2
+            lineNumbers=true
+          )}}
+      </div>
+    </div>
+    <div class="content is-associative">
+      <button class="button is-primary {{if plan.isRunning "is-loading"}}" onclick={{perform plan}} data-test-plan>Plan</button>
+    </div>
+  {{/if}}
+
+  {{#if (eq stage "plan")}}
+    {{#if showPlanMessage}}
+      <div class="notification is-info">
+        <div class="columns">
+          <div class="column">
+            <h3 class="title is-4" data-test-plan-help-title>Job Plan</h3>
+            <p data-test-plan-help-message>This is the impact running this job will have on your cluster.</p>
+          </div>
+          <div class="column is-centered is-minimum">
+            <button class="button is-info" onclick={{toggle-action "showPlanMessage" this}} data-test-plan-help-dismiss>Okay</button>
+          </div>
+        </div>
+      </div>
+    {{/if}}
+    <div class="boxed-section">
+      <div class="boxed-section-head">Job Plan</div>
+      <div class="boxed-section-body is-dark">
+        {{job-diff data-test-plan-output diff=planOutput}}
+      </div>
+    </div>
+    <div class="content is-associative">
+      <button class="button is-primary {{if submit.isRunning "is-loading"}}" onclick={{perform submit}} data-test-run>Run</button>
+      <button class="button is-light" onclick={{action reset}} data-test-cancel>Cancel</button>
```

Comment:
Need some type=button for these two - And not sure if you care about disabling the run button since it's an e-c task, but might be nice to add the `disabled` prop.

Reply:
type=button 🤦‍♂️ HTML can be ridiculous at times.

Good call on the disabled prop.